packages/coinjoin/src/client/CoinjoinRound.ts

```diff
@@ -198,18 +186,78 @@ export class CoinjoinRound extends EventEmitter implements SerializedCoinjoinRou
         return this;
     }
 
-    private processPhase(_accounts: Account[]) {
+    private processPhase(accounts: Account[], prison: CoinjoinPrison) {
         this.lock = createRoundLock(this.options.signal);
+        const processOptions = { ...this.options, signal: this.lock.signal };
         // try to run process on CoinjoinRound
-        return new Promise(resolve => setTimeout(() => resolve(this), 2000));
+        if (this.phase === RoundPhase.InputRegistration) {
+            return inputRegistration(this, prison, processOptions);
+        }
+        if (this.phase === RoundPhase.ConnectionConfirmation) {
+            return connectionConfirmation(this, processOptions);
+        }
+        if (this.phase === RoundPhase.OutputRegistration) {
+            return outputRegistration(this, accounts, prison, processOptions);
+        }
+        if (this.phase === RoundPhase.TransactionSigning) {
+            return transactionSigning(this, processOptions);
+        }
+        if (this.phase === RoundPhase.Ended) {
+            return Promise.resolve(this);
+        }
+        return Promise.resolve(this);
```

Comment:
This would look better as a `switch`. Also, should the `default` case do the same thing as `RoundPhase.Ended`? Shouldn't we somehow log that there's some unexpected round phase?
```suggestion
        switch (this.phase) {
            case RoundPhase.InputRegistration:
                return inputRegistration(this, prison, processOptions);
            case RoundPhase.ConnectionConfirmation:
                return connectionConfirmation(this, processOptions);
            case RoundPhase.OutputRegistration:
                return outputRegistration(this, accounts, prison, processOptions);
            case RoundPhase.TransactionSigning:
                return transactionSigning(this, processOptions);
            case RoundPhase.Ended:
            default:
                return Promise.resolve(this);
        }
```

Reply:
[732481b](https://github.com/trezor/trezor-suite/pull/6712/commits/732481b70daf5567bd5b23c089c3b2a292284be5)